src/cpp/include/openvino/genai/tokenizer.hpp

```diff
@@ -35,16 +35,16 @@ class OPENVINO_GENAI_EXPORTS Tokenizer {
     * @brief encode a single prompt
     * @return pair of [input_ids, attention_mask]
     */
-    TokenizedInputs encode(const std::string prompt);
+    TokenizedInputs encode(const std::string prompt, bool add_special_tokens = true);
```

Comment:
should it be AnyMap with properties? I suppose we can have more parameters in the future and they can be passed in kwargs way?

Reply:
I think it's unlikely that we will add new arguments here in near future.

Can we leave in this PR as is? Since adding properties will requre adding one more overload while we already have 4 overloads (for string, 2 for vectors, initialize_list) and it will look like overkill at the moment when we don't have cases with more that 2 args.

Reply:
we can use std::variant to hide multiple combinations of input parameters

Reply:
Discussed on genai. I agreed to add ov::Properties

Reply:
Implemented with ov::Properties